Sources/App/Core/Dependencies/EnvironmentClient.swift

```diff
@@ -44,6 +44,16 @@ struct EnvironmentClient {
     var mastodonCredentials: @Sendable () -> Mastodon.Credentials?
     var mastodonPost: @Sendable (_ client: Client, _ post: String) async throws -> Void
     var random: @Sendable (_ range: ClosedRange<Double>) -> Double = { XCTFail("random"); return Double.random(in: $0) }
+
+    enum FailureMode: String {
+        case fetchMetadataFailed
+        case findOrCreateRepositoryFailed
+        case invalidURL
+        case noRepositoryMetadata
+        case repositorySaveFailed
+        case repositorySaveUniqueViolation
+    }
+    var shouldFail: @Sendable (_ failureMode: FailureMode) -> Bool = { _ in false }
```

Comment:
I've added this error triggering mechanism, which allows us to configure an environment variable `FAILURE_MODE` as for example

```json
{
    "fetchMetadataFailed": 0.1,
    "findOrCreateRepositoryFailed": 0,
    "invalidURL": 0,
    "noRepositoryMetadata": 0,
    "repositorySaveFailed": 0,
    "repositorySaveUniqueViolation": 0
}
```

which will trigger a `fetchMetadataFailed` error at a rate of 10%.

I'm using this to ensure on `dev` that there's no knock-on effect from errors in subsequent stages. It's also made it easier to ensure locally that the error logs are ok.

We can revert this after testing.